### PR TITLE
BDOG-2353 add service relationships to make displaying inbound and outbound services easier in the catalogue

### PR DESCRIPTION
**Pull request by jordanrowe**

### CODE REVIEW

File: app/uk/gov/hmrc/serviceconfigs/service/ServiceRelationshipService.scala [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    } yield ServiceRelationships(inbound, outbound)
+
+  def updateServiceRelationships(): Future[Unit] =
+    for {
+      slugInfos       <- slugInfoRepository.getAllLatestSlugInfos
+      (srs, failures) <- slugInfos.toList.foldMapM[Future, (List[ServiceRelationship], List[String])]( slugInfo =>
+                           serviceRelationshipsFromSlugInfo(slugInfo)
+                             .map { res => (res.toList, List.empty[String]) }
+                             .recover { case NonFatal(e) =>
+                               logger.warn(s"Error encountered when getting service relationships for ${slugInfo.name}", e)
```
Comment (Shnick) [inline app/uk/gov/hmrc/serviceconfigs/service/ServiceRelationshipService.scala:53]: Should this be logger.error ?

File: app/uk/gov/hmrc/serviceconfigs/service/ServiceRelationshipService.scala [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+                               (List.empty[ServiceRelationship], List(slugInfo.name))
+                             }
+                         )
+      _               <- if(srs.nonEmpty)
+                           serviceRelationshipsRepository.putAll(srs)
+                         else Future.successful(())
+      _               =  if(failures.nonEmpty)
+                           logger.warn(s"Failed to update service relationships for ${failures.mkString(", ")}")
```
Comment (Shnick) [inline app/uk/gov/hmrc/serviceconfigs/service/ServiceRelationshipService.scala:61]: Same again

File: app/uk/gov/hmrc/serviceconfigs/service/ConfigService.scala [hunk 3]
```diff
@@ -243,12 +243,22 @@ class ConfigService @Inject()(
 
   def appConfig(serviceName: String)(implicit hc: HeaderCarrier): Future[Seq[ConfigSourceEntries]] =
```
Comment (colin-lamed) [inline app/uk/gov/hmrc/serviceconfigs/service/ConfigService.scala:244]: can we remove this variant and the endpoint now? It was added for the outbound routes, which has now been reworked.

**Review by colin-lamed — APPROVED** [review]

File: app/uk/gov/hmrc/serviceconfigs/service/ServiceRelationshipService.scala (hunk 1 above)
Comment (jordanrowe) [reply app/uk/gov/hmrc/serviceconfigs/service/ServiceRelationshipService.scala:53]: I think warn is more appropriate in this context as it's not going to cause the whole job to fail

**Pull request merged**